src/linkml_map/functions/unit_conversion.py

```diff
@@ -79,28 +79,32 @@ def convert_units(
     :param to_unit:
     :return: converted magnitude
     """
-    import pint
-
     ureg: pint.UnitRegistry = get_unit_registry(system)
     from_unit = normalize_unit(from_unit, system)
     to_unit = normalize_unit(to_unit, system)
     try:
         from_unit_q = ureg.parse_units(from_unit)
-    except lark.exceptions.UnexpectedCharacters:
-        raise UndefinedUnitError(f"Cannot parse unit: {from_unit}")
-    except pint.errors.UndefinedUnitError:
-        raise UndefinedUnitError(f"Unknown source unit: {from_unit}")
+    except lark.exceptions.UnexpectedCharacters as err:
+        msg = f"Cannot parse unit: {from_unit}"
+        raise UndefinedUnitError(msg) from err
```

Comment:
add in the `as err` / `from err` to allow the call stack to be correctly reported